src/recap.c

```diff
@@ -0,0 +1,490 @@
+  struct pcap_pkthdr32 pkt_header;
+};
+
+void show_app_version(void) {
+  char buf[10];
+
+  snprintf(buf, ARRAY_SIZE(buf), "%d.%d.%d", RECAP_VERSION_MAJOR,
+           RECAP_VERSION_MINOR, RECAP_VERSION_PATCH);
+  fprintf(stdout, "recap app version %s\n", buf);
+}
+
+void show_app_help(char *app_name) {
+  show_app_version();
+  fprintf(stdout, "Usage:\n");
+  fprintf(stdout, USAGE_STRING, basename(app_name));
+  fprintf(stdout, "%s", description_string);
+  fprintf(stdout, "\nOptions:\n");
+  fprintf(stdout, "\t-p filename\t Path to the pcap file name\n");
+  fprintf(stdout, "\t-f filename\t Path to the capture db\n");
+  fprintf(stdout, "\t-i interface\t Interface name to save to db\n");
+  fprintf(stdout,
+          "\t-d\t\t Verbosity level (use multiple -dd... to increase)\n");
+  fprintf(stdout, "\t-h\t\t Show help\n");
+  fprintf(stdout, "\t-v\t\t Show app version\n\n");
+  fprintf(stdout, "Copyright NQMCyber Ltd\n\n");
+  exit(EXIT_SUCCESS);
+}
+
+/* Diagnose an error in command-line arguments and
+   terminate the process */
+void log_cmdline_error(const char *format, ...) {
+  va_list argList;
+
+  fflush(stdout); /* Flush any pending stdout */
+
+  fprintf(stdout, "Command-line usage error: ");
+  va_start(argList, format);
+  vfprintf(stdout, format, argList);
+  va_end(argList);
+
+  fflush(stderr); /* In case stderr is not line-buffered */
+  exit(EXIT_FAILURE);
+}
+
+void process_app_options(int argc, char *argv[], uint8_t *verbosity,
+                         char **pcap_path, char **db_path, char **ifname) {
+  int opt;
+
+  while ((opt = getopt(argc, argv, OPT_STRING)) != -1) {
+    switch (opt) {
+      case 'h':
+        show_app_help(argv[0]);
+        break;
+      case 'v':
+        show_app_version();
+        exit(EXIT_SUCCESS);
+        break;
+      case 'p':
+        *pcap_path = os_strdup(optarg);
+        break;
+      case 'f':
+        *db_path = os_strdup(optarg);
+        break;
+      case 'i':
+        *ifname = os_strdup(optarg);
+        break;
+      case 'd':
+        (*verbosity)++;
+        break;
+      case ':':
+        log_cmdline_error("Missing argument for -%c\n", optopt);
+        break;
+      case '?':
+        log_cmdline_error("Unrecognized option -%c\n", optopt);
+        break;
+      default:
+        show_app_help(argv[0]);
+    }
+  }
+}
+
+ssize_t read_pcap_stream_fd(struct pcap_stream_context *pctx, size_t len,
+                            char **data) {
+  if ((*data = os_malloc(len)) == NULL) {
+    log_errno("os_malloc");
+    return -1;
+  }
+
+  return (ssize_t)fread(*data, sizeof(char), len, pctx->pcap_fd);
+}
+
+ssize_t read_pcap(struct pcap_stream_context *pctx, size_t len) {
+  char *data = NULL;
+  ssize_t read_size, current_size;
+
+  if ((read_size = read_pcap_stream_fd(pctx, len, &data)) < 0) {
+    log_error("read_pcap_stream_fd fail");
+    return -1;
+  }
+
+  pctx->total_size += read_size;
+
+  current_size = read_size + pctx->data_size;
```

Comment:
Declare variable where it's initialized/used.

```suggestion
  ssize_t read_size = read_pcap_stream_fd(pctx, len, &data);
  if (read_size < 0) {
    log_error("read_pcap_stream_fd fail");
    return -1;
  }

  pctx->total_size += read_size;

  ssize_t current_size = read_size + pctx->data_size;
```